server.py

```diff
@@ -26,31 +26,19 @@ def index():
 def add_user():
     """Add a new user to user database with information provided. Send to user
        homepage via /login"""
-
-    first_name = request.form.get('first_name')
-    last_name = request.form.get('last_name')
-    email = request.form.get('email')
-    password = request.form.get('password')
-
-    QUERY = User.query.filter_by(email=email).first()
+    user_data = dict(request.form)
+    QUERY = User.query.filter_by(email=user_data.get('email')).first()
 
     if QUERY is not None:
         flash('User already exists. Please log-in.')
         return redirect('/')
-
     else:
         # Add user to database.
-        user = User(fname=first_name, lname=last_name, email=email,
-                    password=password)
-        db.session.add(user)
-        db.session.commit()
+        user = User(**user_data)
+        user.save()
 
         # Update session dictionary.
-        session['current_user'] = {'user_id': user.user_id,
-                                   'email': user.email,
-                                   'password': user.password,
-                                   'first_name': user.fname,
-                                   'last_name': user.lname}
+        session['current_user'] = {'user_id': user.user_id}
```

Comment:
Never put PII (personally identifying information) in the client. Even though session is encrypted, there's no point in time where you need to display the users password, for example, in the client. So just don't return this sort of information at all. It's best practice.